Code-Code/TypePrediction-TypeScript/README.md

```diff
@@ -0,0 +1,121 @@
+
+
+### Dataset
+
+The dataset originates from the MSR '22 paper [ManyTypes4TypeScript: A Comprehensive TypeScript Dataset for Sequence-Based Type Inference](). The dataset is available for download on Zenodo. 
+
+### Download and Preprocess
+
+1.Download dataset either using wget per below or by clicking on the [link](https://zenodo.org/record/6387001/files/ManyTypes4TypeScript.tar.gz). Unzip and run the preprocess script with chosen parameters like type vocabulary size. To use the default type vocabulary size of 50k types, train#.jsonl, test.jsonl, valid.jsonl are already available in the main directory.
+
+```shell
+cd CodeXGLUE/Code-Code/TypePrediction-TypeScript/dataset
+wget https://zenodo.org/record/6336114/files/ManyTypes4TypeScript.tar.gz -O ManyTypes4TypeScript.tar.gz
+tar -xvzf ManyTypes4TypeScript.tar.gz
+
+```
+
+2.Preprocess dataset
+Use the process_datafiles.py in dataset folder.
+
+```shell
+python3 process_datafiles.py <vocab-size> 
+```
+A vocabulary of 50,000 was used in ManyTypes4TypeScript paper.
+
+### Data Format
+
+After preprocessing dataset there should be five jsonl files. train0.jsonl, train1.jsonl, train2.jsonl, test.jsonl, valid.jsonl. Train is split to accomodate Git LFS.
+
+Each line represents a file. The data fields are the same among all splits.
+
+|field name. | type        |               description                  |
+|------------|-------------|--------------------------------------------|
+|tokens      |list[string] | Sequence of tokens (word tokenization)     |
+|labels      |list[string] | A list of corresponding types              |
+|url         |string       | Repository URL                             |
+|path        |string       | Original file path that contains this code |
+|commit_hash |string       | Commit identifier in the original project  |
+|file        |string       | File name                                  |
+
+### Data Splits
+
+|   name   |  train   |test|   validation  |
+|---------:|---------:|---------:|--------:|
+|projects  |  11,413 (81.8%) | 1,336 (9.58%)  | 1,204 (8.62%) |
+|files     |  486,477 (90.16%) |28,045 (5.20%)|  25,049 (4.64%)|
+|sequences |  1,727,927 (91.95%)| 81,627 (4.34%)| 69,652 (3.71%)  |
+|types     |  8,696,679 (95.33%) |  224,415  (2.46%)|  201,428 (2.21%) |
+
+## Evaluator
+
+We provide a script to evaluate predictions for this task, and report accuracy score. It can be found at evaluator/evaluator.py.
+The train.py in the code directory also has a built in evaluator and reports scores before generating a list of predictions.
+
+### Example
+
+```shell
+python evaluator/evaluator.py -a evaluator/gold_labels.txt -p evaluator/example-codebert-predictions.txt
+```
+{'Acc': 0.6280417975625515}
+
+### Input predictions
+
+A prediction file that has predictions in the form of 
+```shell
+index	type
+```
+See example-codebert-predictions.txt.
+
+## Pipeline-CodeBERT
```

Comment:
It would be better to list required python packages here.

Reply:
Cleaned up package list and added a requirements.txt. Will be in new commit.